src/communication/v2/session.hpp

```diff
@@ -253,23 +272,26 @@ class WebsocketSession : public std::enable_shared_from_this<WebsocketSession<TS
  * Sessions. It handles socket ownership, inactivity timeout and protocol
  * wrapping.
  */
-template <typename TSession, typename TSessionData>
-class Session final : public std::enable_shared_from_this<Session<TSession, TSessionData>> {
+template <typename TSession, typename TSessionContext>
+class Session final : public std::enable_shared_from_this<Session<TSession, TSessionContext>> {
   using TCPSocket = tcp::socket;
   using SSLSocket = boost::asio::ssl::stream<TCPSocket>;
-  using std::enable_shared_from_this<Session<TSession, TSessionData>>::shared_from_this;
+  using std::enable_shared_from_this<Session<TSession, TSessionContext>>::shared_from_this;
 
  public:
   template <typename... Args>
   static std::shared_ptr<Session> Create(Args &&...args) {
     return std::shared_ptr<Session>(new Session(std::forward<Args>(args)...));
   }
 
+#ifdef MG_ENTERPRISE
+  ~Session() { session_context_->Delete(session_); }
+
   Session(const Session &) = delete;
   Session(Session &&) = delete;
   Session &operator=(const Session &) = delete;
   Session &operator=(Session &&) = delete;
-  ~Session() = default;
+#endif
```

Comment:
Why do we put copy/move constructors/operators only for MG_ENTERPRISE?

Reply:
I was modifying the ~Session and made a mistake.
They should be deleted in any case.